src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.Types.cs

```diff
@@ -46,5 +46,38 @@ public struct IntPtrAndHandle
             [FieldOffset(0)]
             internal RuntimeTypeHandle typeHandle;
         }
+
+        // keep in sync with types\internal.ts
+        public enum MainThreadingMode : int
+        {
+            // Running the managed main thread on UI thread.
+            // Managed GC and similar scenarios could be blocking the UI.
+            // Easy to deadlock. Not recommended for production.
+            UIThread = 0,
+            // Running the managed main thread on dedicated WebWorker. Marshaling all JavaScript calls to and from the main thread.
+            DeputyThread = 1,
+        }
+
+        // keep in sync with types\internal.ts
+        public enum JSThreadBlockingMode : int
+        {
+            // throw PlatformNotSupportedException if blocking .Wait is called on threads with JS interop, like JSWebWorker and Main thread.
+            // Avoids deadlocks (typically with pending JS promises on the same thread) by throwing exceptions.
+            NoBlockingWait = 0,
+            // allow .Wait on all threads.
+            // Could cause deadlocks with blocking .Wait on a pending JS Task/Promise on the same thread or similar Task/Promise chain.
+            AllowBlockingWait = 100,
+        }
+
+        // keep in sync with types\internal.ts
+        public enum JSThreadInteropMode : int
+        {
+            // throw PlatformNotSupportedException if synchronous JSImport/JSExport is called on threads with JS interop, like JSWebWorker and Main thread.
+            // calling synchronous JSImport on thread pool or new threads is allowed.
+            NoSyncJSInterop = 0,
+            // allow non-re-entrant synchronous blocking calls to and from JS on JSWebWorker on threads with JS interop, like JSWebWorker and Main thread.
+            // calling synchronous JSImport on thread pool or new threads is allowed.
+            SimpleSynchronousJSInterop = 1,
```

Comment:
I think we should come with a better name (before api review?)

Reply:
@lewing also didn't like it. Please come up with suggestions.

Reply:
I'm voting for loger and more descriptive name, "simple" has too much meanings

Reply:
- SingleSynchronousCallChainJSInterop
- SingleSynchronousCallJSInterop

Reply:
`SingleSynchronousCallJSInterop` is not bad, I like it

Reply:
BTW: those would not be visible on C# API, just JS config